src/main/java/org/broadinstitute/hellbender/tools/exome/eval/EvaluateCopyNumberTriStateCalls.java

```diff
@@ -0,0 +1,772 @@
+     * @return never {@code null}.
+     */
+    private VariantEvaluationContext composeNonTruthOverlappingVariantContext(final VariantContext call, final TargetCollection<Target> targets) {
+        final VariantEvaluationContextBuilder result = new VariantEvaluationContextBuilder();
+        result.loc(call.getContig(), call.getStart(), call.getEnd());
+        result.attribute(VCFConstants.END_KEY, call.getEnd());
+        result.id(call.getID());
+        result.alleles(call.getAlleles());
+        result.genotypes(
+                call.getGenotypes().stream()
+                .map(g -> composeNonTruthOverlappingGenotype(call, g))
+                .collect(Collectors.toList()));
+        result.attribute(GenotypeCopyNumberTriStateSegments.NUMBER_OF_TARGETS_KEY, targets.targetCount(call));
+        result.evidence(null, Collections.singletonList(call));
+        return result.make();
+    }
+
+    private Genotype composeNonTruthOverlappingGenotype(final VariantContext enclosingContext, final Genotype genotype) {
+        final GenotypeBuilder builder = new GenotypeBuilder(genotype.getSampleName());
+        GATKProtectedVariantContextUtils.setGenotypeQualityFromPLs(builder, genotype);
+
+        final int[] PL = genotype.getPL();
+        final int callAlleleIndex = GATKProtectedMathUtils.minIndex(PL);
+        final double quality = callQuality(genotype);
+        builder.alleles(Collections.singletonList(enclosingContext.getAlleles().get(callAlleleIndex)));
+        builder.attribute(VariantEvaluationContext.CALL_QUALITY_KEY, quality);
+        final boolean discovered = GenotypeCopyNumberTriStateSegments.DISCOVERY_TRUE.equals(
+                GATKProtectedVariantContextUtils.getAttributeAsString(genotype, GenotypeCopyNumberTriStateSegments.DISCOVERY_KEY,
+                        GenotypeCopyNumberTriStateSegments.DISCOVERY_FALSE));
+        if (callAlleleIndex != 0 && discovered) {
+            builder.attribute(VariantEvaluationContext.EVALUATION_CLASS_KEY, EvaluationClass.UNKNOWN_POSITIVE.acronym);
+        }
+        if (quality < filterArguments.minimumCalledSegmentQuality) {
+            builder.filter(EvaluationFilter.LowQuality.acronym);
+        } else {
+            builder.filter(EvaluationFilter.PASS);
+        }
+        return builder.make();
+    }
+
+    private double callQuality(final Genotype genotype) {
+        final int[] pls = genotype.getPL();
+        int bestPL = pls[0];
+        int secondBestPL = Integer.MAX_VALUE;
+        for (int i = 1; i < pls.length; i++) {
+            if (pls[i] < bestPL) {
+                secondBestPL = bestPL;
+                bestPL = pls[i];
+            } else if (pls[i] < secondBestPL) {
+                secondBestPL = pls[i];
+            }
+        }
+        return secondBestPL - bestPL;
+    }
+
+    private VariantEvaluationContext composeTruthOverlappingVariantContext(final VariantContext truth, final List<VariantContext> calls,
```

Comment:
Is there any way to reduce the code duplication between the truth-overlapping and the non-truth-overlapping methods?